walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -270,5 +360,58 @@ public void notEnoughRoom() {
 
     Assert.assertEquals(expected, actual);
   }
+
+  @Test
+  public void addOptionalAttendeeConsidered() {
+    // Events           |--A--|     |--B--|     |--C--|
+    // Day     : |------------------------------------------|
+    // Options : |--1--|     |--2--|     |--3--|      |--4--|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartDuration(TIME_0100PM, DURATION_2_HOUR),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 3", TimeRange.fromStartDuration(TIME_0700PM, DURATION_1_HOUR),
+            Arrays.asList(PERSON_C)));
+
+    MeetingRequest request =
+        new MeetingRequest(NO_ATTENDEES, DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_A);
+    request.addOptionalAttendee(PERSON_B);
+    request.addOptionalAttendee(PERSON_C);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0830AM, TIME_0100PM, false),
+            TimeRange.fromStartEnd(TIME_0300PM, TIME_0700PM, false),
+            TimeRange.fromStartEnd(TIME_0800PM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void addOptionalAttendeeConsidered2() {
+    // Events    |--------A--------||----------B------------|
+    // Day     : |------------------------------------------|
+    // Options : |--1--|     |--2--|     |--3--|      |--4--|
```

Comment:
This picture doesn't look correct I think?  If your expected outcome below is WHOLE_DAY, that would only be 1 item, vs. the 4 you have on this image.